internal/indexnode/task.go

```diff
@@ -356,18 +356,23 @@ func (it *indexBuildTask) BuildIndex(ctx context.Context) error {
 }
 
 func (it *indexBuildTask) SaveIndexFiles(ctx context.Context) error {
+
+	gcIndex := func() {
+		if err := it.index.Delete(); err != nil {
+			log.Ctx(ctx).Error("IndexNode indexBuildTask Execute CIndexDelete failed", zap.Error(err))
```

Comment:
what happen if delete fail, should we just abort?

Reply:
`index.Delete` fails when:
- Double Closed, which need no retry
- CGO call failured, in which case cpp pointer has some issue, retrying cannot fix this case